securedrop/tests/test_source.py

```diff
@@ -28,6 +28,11 @@ def setUp(self):
     def tearDown(self):
         utils.env.teardown()
 
+    def test_page_not_found(self):
+        response = self.client.get('/UNKNOWN')
+        self.assertEqual(response.status_code, 404)
```

Comment:
Flask-Testing provides some handy methods for the web application context. E.g., you could use `self.assert404(response)` here.